### PR TITLE
Add checkpoint/restore support to the execdriver

### PR DESCRIPTION
**Pull request by boucher**
This PR also includes an individual commit that brings the libcontainer vendor up to date. I'm not sure what the protocol is, but I can separate that out into a separate pull request if desired (but obviously its necessary for this patch).

### CODE REVIEW

File: daemon/execdriver/native/driver.go [hunk 1]
```diff
@@ -274,6 +274,144 @@ func (d *driver) Unpause(c *execdriver.Command) error {
...
+		d.cleanContainer(c.ID)
+	}()
+
+
+	// Since the CRIU binary exits after restoring the container, we
+	// need to reap its child by setting PR_SET_CHILD_SUBREAPER (36)
+	// so that it'll be owned by this process (Docker daemon) after restore.
+	//
+	// XXX This really belongs to where the Docker daemon starts.
+	if _, _, syserr := syscall.RawSyscall(syscall.SYS_PRCTL, 36, 1, 0); syserr != 0 {
```
Comment (boucher) [inline daemon/execdriver/native/driver.go]: @avagin Is this still needed now that we're using criu/swrk?

Comment (avagin) [inline daemon/execdriver/native/driver.go]: No, this isn't needed, because we execute criu with rst_sibling.

**Comment by duglin** [comment]
I believe all of the changes under "vendor/*" will need to be done under PRs in those particular projects since docker just copies in their code.

**Comment by boucher** [comment]
The changes to vendor are already in libcontainer.

**Comment by duglin** [comment]
Then i would think we should pick up the libcontainer update separately. As you suggested in your first comment.

**Comment by boucher** [comment]
@crosbymichael ping

**Comment by LK4D4** [comment]
@boucher Something with tags I think, at least for windows.

**Comment by boucher** [comment]
Replaced with this PR: https://github.com/docker/docker/pull/13602

**Pull request closed without merging**